src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -194,6 +194,18 @@ bool convert_precision(ov::pass::PassBase& pass,
             bool is_output_precision_changed = false;
 
             for (auto& node : ops) {
+                // Convert node should avoid converting the output element type from boolean to u8, if it connects to a
```

Comment:
Current approach might negatively affect other plugin which doesn't support Bool precision at all.
We need to understand it is possible to find more generic solution that will keep correct math after precision conversion.
cc'ed @itikhono

Reply:
@xuchen-intel I agree with Dmitry, we need to implement a more generic solution.
The new idea is to use `additional_fuse_map` argument of ConvertPrecision ctor
![image](https://user-images.githubusercontent.com/21181928/203992364-d74d90a2-bd90-4371-a4d7-913cb7886b6d.png)
and pass a new "fuse precision" function  via this argument to ConvertPrecision transformation and replace an existing function (`{opset4::Convert::get_type_info_static(), fuse_type_to_convert},`) in type_to_fuse map. 
To re-write the existing function we need to change this line
![image](https://user-images.githubusercontent.com/21181928/203992708-43396002-ad25-4f99-91f3-8d0baaedf879.png)
A new function should contain the next logic:
![image](https://user-images.githubusercontent.com/21181928/203994301-fdc228a8-3850-444f-a523-74bc82d5e98c.png)
Could you implement this solution?

Reply:
Great suggestion! Applied. Thanks Ivan and Dmitry!